nncf/experimental/onnx/algorithms/min_max_quantization.py

```diff
@@ -0,0 +1,163 @@
+        super().__init__(statistics_collector, parameters)
+        self._weight_quantizers = []
+        self._activation_quantizers = []
+        self.global_quantizer_constraints = {}
+
+    def generate_stat_collector(self, quantizer_config: QuantizerConfig):
+        if self.range_type == 'min_max':
+            is_symmetric = True if quantizer_config.mode == QuantizationMode.SYMMETRIC else False
+            if quantizer_config.per_channel:
+                axes = (0, 2, 3)
+                return ONNXMinMaxStatisticCollector(is_symmetric, axes)
+            return ONNXMinMaxStatisticCollector(is_symmetric, None)
+
+    def _create_model_transformer(self, model: onnx.ModelProto):
+        self.model_transformer = ONNXModelTransformer(model)
+
+    def _get_quantizer_setup(self, model: onnx.ModelProto):
+        nncf_graph = GraphConverter.create_nncf_graph(model)
+        ip_graph = InsertionPointGraph(nncf_graph)
+        pattern = ONNX_HW_FUSED_PATTERNS.get_full_pattern_graph()
+        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
+
+        weight_nodes = nncf_graph.get_nodes_by_metatypes(QUANTIZATION_LAYER_METATYPES)
+        quantizable_layer_nodes = [QuantizableWeightedLayerNode(weight_node, [QuantizerConfig()]) for weight_node in
+                                   weight_nodes]
+
+        hw_config_type = HWConfigType.from_str(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[self.target_device])
+        hw_config_path = ONNXHWConfig.get_path_to_hw_config(hw_config_type)
+        hw_config = ONNXHWConfig.from_json(hw_config_path)
+
+        solver = QuantizerPropagationSolver(ignored_scopes=self.ignored_scopes,
+                                            hw_config=hw_config,
+                                            default_trait_to_metatype_map=DEFAULT_ONNX_QUANT_TRAIT_TO_OP_DICT,
+                                            default_qconfig_list=[self._get_default_qconfig()],
+                                            quantizable_layer_nodes=quantizable_layer_nodes,
+                                            quantize_outputs=self.quantize_outputs)
+
+        quantization_proposal = solver.run_on_ip_graph(ip_graph)
+        multi_config_setup = quantization_proposal.quantizer_setup
+        single_config_setup = multi_config_setup.select_first_qconfig_for_each_point()
+        finalized_proposal = quantization_proposal.finalize(single_config_setup)
+        final_setup = solver.get_final_quantizer_setup(finalized_proposal)
+        return final_setup
+
+    def apply(self, model: onnx.ModelProto, engine: ONNXEngine):
+        modified_model = modify_onnx_model_for_quantization(model)
+        self._create_model_transformer(modified_model)
+        transformation_layout = ONNXTransformationLayout()
+        transformation_commands = []
+        onnx_graph = ONNXGraph(modified_model)
+
+        # If statistics were not collected
+        if not self.statistics_collector.layers_statistics:
```

Comment:
Why the algorithm should be responsible for collecting statistics, in the case of a simple algorithm like MinMaxQuantization?

Reply:
In the simplest scenario, MinMaxQuantization is used outside or without PostTrainingQuantization. In that case, the algorithm should collect statistics by itself. While PostTrainingQuantization should collect statistics for several algorithms (MinMaxQuantization, BiasCorrection) at one time, and then algorithms use it next.
So, MinMaxQuantization should support both cases, because of that I first check if the statistics were collected if it is not the algorithm collects them.

Reply:
Not sure that algorithms should collect statistics themself. The POT pipeline calls the statistics collector for each algorithm. Also, in the case of MinMax + BiasCorrection can be represented not as part of PostTrainingQuantization, but separately. And using the current pipeline, PostTrainingQuantization will be executed faster than MinMax + BiasCorrection, but the result will be the same. I'm not sure that this behavior is correct because in the POT DefaultQuantization (ChannelAlignment + MinMax + BiasCorrection) takes the same time that ChannelAlignment + MinMax + BiasCorrection but without the DQ wrapper.

Reply:
Nice comment! I didn't think about comparing the perf between PostTrainingQuantizaton and the contiguous execution of the algos behind it.
So the time would be different as you mentioned and it is good, I suppose. But this step could lead to issues because of increasing the complexity level of the algos.
This is a point for the discussion.
@AlexKoff88, may you please describe your opinion. Thanks!

Reply:
After the discussion with @KodiaqQ and @AlexKoff88 we decided to move the statistics calculation from the algorithm logic. 
Now the statistics calculation is done by CompressionBuilder.